scripts/open-coredump.sh

```diff
@@ -0,0 +1,130 @@
+#!/bin/bash
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2022-present ScyllaDB
+#
+
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+
+# Script for "one-click" opening of coredumps.
+# It extracts the build-id from the coredump, retrieves metadata for that
+# build, downloads the binary package, the source code and finally
+# launches the dbuild container, with everything ready to load the coredump.
+# The script is reentrant: running it after the prepartory steps will re-use what
+# is already donwloaded. It is *strongly* recommended to run this from an empty
+# directory, with nothing but the core-file present.
+#
+# Requirements:
+# * eu-unstrip
+# * curl
+# * jq
+# * git
+
+set -e
+
+SCYLLA_S3_RELOC_SERVER_URL="http://backtrace.scylladb.com"
+
+if [[ $# != 1 ]]; then
+	echo Please provide the path to the coredump
+	exit 1
+fi
+
+for required in eu-unstrip jq curl git; do
+	if ! type $required >& /dev/null; then
+		echo Please install $required first
+		exit 1
+	fi
+done
+
+COREFILE=$1
+
+BUILD_ID=$(eu-unstrip -n --core ${COREFILE} | cut -f2 -d' ' | cut -f1 -d@)
+
+echo "Build id: ${BUILD_ID}"
+
+BUILD=$(curl -s -X GET "${SCYLLA_S3_RELOC_SERVER_URL}/build.json?build_id=${BUILD_ID}")
+
+RESPONSE_BUILD_ID=$(jq -r .build_id <<< $BUILD)
+VERSION=$(jq -r .version <<< $BUILD)
+PRODUCT=$(jq -r .product <<< $BUILD)
+RELEASE=$(jq -r .release <<< $BUILD)
+ARCH=$(jq -r .arch <<< $BUILD)
+BUILD_MODE=$(jq -r .build_mode <<< $BUILD)
+PACKAGE_URL=$(jq -r .package_url <<< $BUILD)
+
+if [[ "$RESPONSE_BUILD_ID" != "$BUILD_ID" ]]
+then
+    echo "Mismatching build id: requested ${BUILD_ID} but got ${RESPONSE_BUILD_ID}";
+    exit 1
+fi
+
+echo "Matching build is ${PRODUCT}-${VERSION} ${RELEASE} ${BUILD_MODE}-${ARCH}"
+
+PACKAGE_FILE=$(basename ${PACKAGE_URL})
+
+if ! [[ -d ${PRODUCT} ]]
+then
+    if ! [[ -f ${PACKAGE_FILE} ]]
+    then
+        echo "Downloading relocatable package from ${PACKAGE_URL}"
+        curl --output ./${PACKAGE_FILE} ${PACKAGE_URL}
+    else
+        echo "Relocatable package ${PACKAGE_URL} already downloaded"
+    fi
+
+    echo "Extracting package ${PACKAGE_FILE}"
+    tar -zxf $PACKAGE_FILE
+else
+    echo "Relocatable package ${PACKAGE_URL} already downloaded and extracted"
```

Comment:
I don't understand why that's coupled with `-d ${PRODUCT}`

Reply:
Yes.  In the scylla reloc server cache I create an "extracted" sub-directory and extract the tar archive there so I can rely on a known name.  For the relocatable package it even changed over time where the scylla sub-directory added in a later stage.  historically IIRC it had only libexec and libreloc directly under the root.